osquery/core/init.cpp

```diff
@@ -368,6 +388,54 @@ Initializer::Initializer(int& argc,
   platformSetup();
 }
 
+Initializer::~Initializer() {}
+
+bool terminateActiveOsqueryInstance() {
+  auto pid_res = Pidfile::read(FLAGS_pidfile);
+  if (pid_res.isError()) {
+    auto error = pid_res.getErrorCode();
+
+    if (error != Pidfile::Error::NotRunning) {
+      LOG(ERROR) << "Failed to read the pidfile: " << error;
+    }
+
+    return false;
+  }
+
+  auto pid = static_cast<int>(pid_res.take());
+  if (pid == PlatformProcess::getCurrentPid()) {
+    return true;
+  }
+
+  PlatformProcess target(pid);
+  int status = 0;
+
+  // The pid is running, check if it is an osqueryd process by name.
+  std::stringstream query_text;
+
+  query_text << "SELECT name FROM processes WHERE pid = " << pid
+             << " AND name LIKE 'osqueryd%';";
+
+  SQL q(query_text.str());
+  if (!q.ok()) {
+    LOG(ERROR) << "Error querying processes: " << q.getMessageString();
+    return false;
+  }
+
+  if (q.rows().size() > 0) {
+    LOG(ERROR) << "Killing osqueryd process: " << pid;
+
+    // Do not use SIGQUIT as it will cause a crash on OS X.
+    status = target.kill() ? 0 : -1;
```

Comment:
Does this mean there's no change for a graceful shutdown?

Reply:
This is not a change, as I have just changed the Pidfile implementation in the original kill procedure. It is still to be considered as a non-graceful shutdown.

On Windows, it will use `TerminateProcess()`, and on Posix systems it will send a `SIGKILL` signal